### ISSUE TITLE
Rust agent uses Option::unwrap() more heavily than necessary

### ISSUE THREAD

**Issue by dgibson** [kata-containers/kata-containers#821]
# Description of problem

The Rust agent code uses Option::unwrap() in many places, which will panic if the Option is a None.  In most (maybe all) of those cases we have code that tests the Option is not a None guarding that, but in many cases there are more elegant ways to handle this avoiding unwrap.

**Comment by jodh-intel** [comment]
Agreed. The agent is long-running and functions should pass back errors to the caller where possible rather than panic-ing by calling `expect()` or `unwrap()` on `Option` or `Result` values.

Note that it isn't always possible (or reasonable) to remove `unwrap()`/`expect()` calls. The following are examples of scenarios where they either have to exist or it makes senses for them to be retained:

| Area | Rationale for permitting |
|-|-|
| In test code (the `tests` module) | Panics will cause the test to fail which is desirable |
| `lazy_static!()` | This magic macro cannot "return" a value as it runs before `main()` |
| `defer!()` | Similar to golang's `defer()` but doesn't allow the use of `?` |
| `tokio::spawn(async move {})` | Cannot currently return a `Result` from an `async move` closure afaik |
| If an explicit test is performed before the `unwrap()` / `expect()` | "Just about ok", but not ideal |

**Comment by dgibson** [comment]
> Agreed. The agent is long-running and functions should pass back errors to the caller where possible rather than panic-ing by calling `expect()` or `unwrap()` on `Option` or `Result` values.

True, but not really the point I was getting at.  Most of the instances of `unwrap()` already check for the "bad" case immediately above, so they're not actually in danger of panicing rather than returning errors.  However, there's usually a more elegant way of doing this.  In particular there are a *lot* of constructs like either:

```
    if XXX.is_some() {
            do_something_with(XXX.unwrap();
    }
```

or

```
   if XXX.is_none() {
       return Err(...);
   }
   do_something_with(XXX.unwrap();
```

Both can be better handled using `if let` or `match`.

> Note that it isn't always possible (or reasonable) to remove `unwrap()`/`expect()` calls. The following are examples of scenarios where they either have to exist or it makes senses for them to be retained:

> If an explicit test is performed before the `unwrap()` / `expect()` 	"Just about ok", but not ideal

That's actually *exactly* the case I'm talking about - there are dozens, maybe hundreds, of these which could be handled with better constructs (generally either `if let` or various `Result` methods).

To your list I'd add cases where Rust makes it awkward to elegantly separate out the error cases.  One I've encountered is:
```
    if (!strong_condition(x)) {
        return Err(...);
    }
    do_something_with(x.unwrap());
```
Where `strong_condition(x)` clearly implies `x.is_some()`.  That can be awkward to rearrange in a Rust-natural way without duplicating the error message, which can get bulky and ugly.  I think it's reasonable to leave `unwrap()` here, at least in the first pass.  (Although, tip, using [match guards](https://doc.rust-lang.org/rust-by-example/flow_control/match/guard.html) can make at least some of these cases nice too)
